lib/app/modules/us_core_r4/clinicalnotes_sequence.rb

```diff
@@ -0,0 +1,211 @@
+
+        assert @actual_type_codes.include?(type_code), "Clinical Notes shall have at least one DocumentReference with type #{type_code}"
+      end
+
+      def test_clinical_notes_diagnostic_report(category_code)
+        skip 'No Clinical Notes appear to be available for this patient. Please use patients with more information.' unless @clinicalnotes_found
+
+        search_params = { 'patient': @instance.patient_id, 'category': category_code }
+
+        reply = get_resource_by_params(versioned_resource_class('DiagnosticReport'), search_params)
+        assert_response_ok(reply)
+        assert_bundle_response(reply)
+
+        resource_count = reply&.resource&.entry&.length || 0
+        @resources_found = true if resource_count.positive?
+
+        assert @resources_found, "Clinical Notes shall have at least one DiagnosticReport with category #{category_code}"
+
+        diagnostic_reports = reply&.resource&.entry&.map { |entry| entry&.resource }
+
+        diagnostic_reports&.each do |diarpt|
+          diarpt&.presentedForm&.select { |attachment| attachment&.url&.present? }&.each do |attachment|
+            assert @attachment_urls.key?(attachment.url), "Attachment #{attachment.url} referenced in DiagnosticReport/#{diarpt.id} but not in any DocumentReference"
+            @attachment_urls[attachment.url][:flag] = true
+          end
+        end
+      end
+
+      test 'Server returns expected results from DocumentReference search by patient+clinicalnotes' do
+        metadata do
+          id '01'
+          link 'https://build.fhir.org/ig/HL7/US-Core-R4/clinical-notes-guidance.html'
+          desc %(
+          )
+          versions :r4
+        end
+
+        search_params = { 'patient': @instance.patient_id, 'category': 'clinical-note' }
+        @attachment_urls = {}
+
+        reply = get_resource_by_params(versioned_resource_class('DocumentReference'), search_params)
+        assert_response_ok(reply)
+        assert_bundle_response(reply)
+
+        resource_count = reply&.resource&.entry&.length || 0
+        @clinicalnotes_found = true if resource_count.positive?
+
+        skip 'No Clinical Notes appear to be available for this patient. Please use patients with more information.' unless @clinicalnotes_found
+
+        document_references = reply&.resource&.entry&.map { |entry| entry&.resource }
+
+        @required_type_codes = Set[
+          '11488-4', # Consultation Note
+          '18842-5', # Dischard Summary
+          '34117-2', # History and physical note
+          '28570-0', # Procedure note
+          '11506-3' # Progress not
+        ]
+
+        @actual_type_codes = Set[]
+
+        document_references&.select { |docref| docref&.type&.coding&.present? }&.each do |docref|
+          docref.type.coding.select { |coding| coding&.system == 'http://loinc.org' && @required_type_codes.include?(coding&.code) }&.each do |coding|
+            @actual_type_codes.add(coding.code)
```

Comment:
It's fine to use `add`, but it's more typical to use the "shovel operator" (it's really just a method rather than an operator): `@actual_type_codes << coding.code`

Reply:
fixed